modules/SerialCommHub/main/serial_communication_hubImpl.cpp

```diff
@@ -0,0 +1,367 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 - 2022 Pionix GmbH and Contributors to EVerest
+
+#include "serial_communication_hubImpl.hpp"
+
+#include <chrono>
+#include <connection/exceptions.hpp>
+#include <date/date.h>
+#include <date/tz.h>
+#include <fmt/core.h>
+#include <modbus/exceptions.hpp>
+#include <modbus/utils.hpp>
+
+#include <typeinfo>
+
+namespace module {
+namespace main {
+
+void serial_communication_hubImpl::init() {
+
+    everest::connection::SerialDeviceConfiguration cfg(config.serial_port);
+    cfg.set_sensible_defaults();
+    cfg.set_baud_rate(baudrate_from_int(config.baudrate));
+    cfg.set_stop_bits(everest::connection::SerialDeviceConfiguration::StopBits::One);
+    cfg.set_data_bits(everest::connection::SerialDeviceConfiguration::DataBits::Bit_8);
+
+    switch (config.parity) {
+    case 0:
+        cfg.set_parity(everest::connection::SerialDeviceConfiguration::Parity::None);
+        break;
+    case 1:
+        cfg.set_parity(everest::connection::SerialDeviceConfiguration::Parity::Odd);
+        break;
+    case 2:
+        cfg.set_parity(everest::connection::SerialDeviceConfiguration::Parity::Even);
+        break;
+    }
+
+    serial_device = std::unique_ptr<everest::connection::SerialDevice>(new everest::connection::SerialDevice(cfg));
+    modbus_connection =
```

Comment:
same here, only pass *serial_device
RTUConnection actually owns a reference to the serial device - therefor it is also importent, that the unique_ptr modbus_connection goes out of scope before serial_device (order of member variable listing is important!)

Reply:
yes, that would all be easier if the modbus and serial library would support a constructor without arguments as well